chain.go

```diff
@@ -37,7 +37,7 @@ type Chain struct {
 }
 
 // NewChain creates and empty chain
-func NewChain() (chain *Chain) {
+func (holo *Holochain) NewChain() (err error) {
```

Comment:
Generally, if we needed WireType and were using the chain as the receiver, I changed it to the holochain.

We could do the same with the dht links and not need bi-directional links h <=> dht. Generally pass holochain and delegate to dht and chain per context.